library/linkedList.js

```diff
@@ -0,0 +1,126 @@
+"use strict";
+
+class Node {
+  /**
+   * creating a node
+   * @param {any} data
+   * @return {node}
+   */
+  constructor(data) {
+    this.data = data;
+    this.next = null;
+  }
+
+  addLinkToBack(data) {
+    /**
+     * adding another link to the back
+     * eg. intial node => Node(data:1,next:null)
+     * new node => Node(data:1, next:(data:1, next:null))
+     * it runs a while loop till .next is null. Then inserts the new node such that .next: new node.
+     * @param {any} data
+     * @return {node}
+     */
+
+    let nextNode = this;
+    while (nextNode.next != null) {
+      nextNode = nextNode.next;
+    }
+    nextNode.next = new Node(data);
+    return this;
+  }
+
+  addLinkToFront(value) {
+    /**
+     * adding a node to the begging of the chain
+     * eg. intial node => Node(data:1, next:(data:2, next:null)) and value = "xyz"
+     * new node => Node(data:"xyz", next:(data:1, next:(data:2, next:null)))
+     * to achive the above, values of this.data and this.next is saved in the newNode => making newNode mimic this.
+     * then at this.data inserting value param and this.next's value to be the new node.
+     * @param {any} value
+     * @return {node}
+     */
+    let newNode = new Node();
+    newNode.data = this.data;
+    newNode.next = this.next;
+    this.data = value;
+    this.next = newNode;
+    return this;
+  }
+
+  addLink(value, positionToInsert) {
+    /**
+     * adding a node to a specific position in the node.
+     * 1st position is considered to be at this.next value
+     * @param {any} value
+     * @param {integer} positionToInsert
+     * @return {node}
+     */
+    if (!Number.isInteger(positionToInsert)) {
+      throw new Error("Provide integer value for position.");
+    }
+    if (positionToInsert === 0) {
+      this.addLinkToFront(value);
+    }
+    let nextNode = this;
+    let currentPosition = 1;
+    while (nextNode.next != null && currentPosition != positionToInsert) {
```

Comment:
notice the duplication of some code between here and add to back,
can we re-use some pieces ?

Reply:
I don't think i can re-use as i don't have currentPosition and positionToInsert variables under addToBack. I can add currentPosition to addToBack, but PositionToInsert will still be missing. If you are not satisfied with my reasoning, please let me know and I will look for alternate solutions to reuse addToBack under addLink(value, positionToInsert) method.